src/femtomes.cpp

```diff
@@ -103,17 +120,97 @@ int GPSDriverFemto::handleMessage(int len)
+        {
+            float heading = _femto_uav_gps.heading;
+            heading *= M_PI_F / 180.0f; // deg to rad, now in range [0, 2pi]
+            heading -= _heading_offset; // range: [-pi, 3pi]
+
+            if (heading > M_PI_F)
+            {
+                heading -= 2.f * M_PI_F; // final range is [-pi, pi]
+            }
+            _gps_position->heading = heading;
+        }
+        else
+        {
+            _gps_position->heading = NAN;
+        }
 
-		_gps_position->heading = heading;
 		_gps_position->timestamp = gps_absolute_time();
 
 		ret = 1;
 	}
+	else if(messageid == FEMTO_MSG_ID_RTCM3)    /**< rtcm3 */
+    {
+        gotRTCMMessage(_rtcm_parsing->message(), _rtcm_parsing->messageLength());
+        decodeInit();
+        ret = 1;
+    }
+	else if(messageid == FEMTO_MSG_ID_GPGGA && (memcmp(_femto_msg.data + 3, "GGA,", 3) == 0)) /**< GPGGA */
+    {
+        int uiCalcComma = 0;
+
+        for (int i = 0 ; i < len; i++)
+        {
+            if (_femto_msg.data[i] == ',') { uiCalcComma++; }
+        }
+        if(uiCalcComma == 14)
+        {
+            char *bufptr = (char *)(_femto_msg.data + 6);
+            char *endp = nullptr;
+            double ashtech_time = 0.0,lat = 0.0 , lon = 0.0 , alt = 0.0;
+            int num_of_sv = 0, fix_quality = 0;
+            double hdop = 99.9;
+            char ns = '?', ew = '?';
+
+            if (bufptr && *(++bufptr) != ',') { ashtech_time = strtod(bufptr, &endp); bufptr = endp; }
+
+            if (bufptr && *(++bufptr) != ',') { lat = strtod(bufptr, &endp); bufptr = endp; }
+
+            if (bufptr && *(++bufptr) != ',') { ns = *(bufptr++); }
+
+            if (bufptr && *(++bufptr) != ',') { lon = strtod(bufptr, &endp); bufptr = endp; }
+
+            if (bufptr && *(++bufptr) != ',') { ew = *(bufptr++); }
+
+            if (bufptr && *(++bufptr) != ',') { fix_quality = strtol(bufptr, &endp, 10); bufptr = endp; }
+
+            if (bufptr && *(++bufptr) != ',') { num_of_sv = strtol(bufptr, &endp, 10); bufptr = endp; }
+
+            if (bufptr && *(++bufptr) != ',') { hdop = strtod(bufptr, &endp); bufptr = endp; }
+
+            if (bufptr && *(++bufptr) != ',') { alt = strtod(bufptr, &endp); bufptr = endp; }
+
+            if (ns == 'S')
+            {
+                lat = -lat;
+            }
+
+            if (ew == 'W')
+            {
+                lon = -lon;
+            }
+
+			FEMTO_UNUSED(ashtech_time)
+			FEMTO_UNUSED(hdop)
+
+            if(!_correction_output_activated && 7 == fix_quality)
+            {
+                lat = (int(lat * 0.01) + (lat * 0.01 - int(lat * 0.01)) * 100.0 / 60.0) * 10000000;
+                lon = (int(lon * 0.01) + (lon * 0.01 - int(lon * 0.01)) * 100.0 / 60.0) * 10000000;
+                alt = alt * 1000;
+
+                activateRTCMOutput();
+                sendSurveyInStatusUpdate(false, true, lat, lon, (float)alt);
+                _correction_output_activated = true;
```

Comment:
Can you set this in `activateRTCMOutput`?

Reply:
Ok, I will set this in activateRTCMOutput.